ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs

```diff
@@ -6,25 +6,52 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using System.Web;
 
 namespace ASPNetCoreMastersTodoList.Api.Controllers
 {
-    public class ItemsController : Controller
+    [Route("{controller}")]
+    [ApiController]
+    public class ItemsController : ControllerBase
     {
-        public int Get(int userId)
+        public readonly IItemService _service;
+        public ItemsController(IItemService service)
         {
-            return new ItemService().GetAll(userId);
+            _service = service;
+        }
+        [HttpGet]
+        public IActionResult GetAll(int userId)
+        {
+            return Ok(_service.GetAll());
+        }
+        [HttpGet("{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+            return Ok(_service.Get(itemId));
+        }
+        [HttpGet("filterBy")]
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
+        {
+            return Ok(_service.GetAllByFilter(new ItemByFilterDTO() { }));
```

Comment:
You can get specific values from filters since it is a dictionary. For example you can get the text property and then map it to your ItemByFilterDTO